scripts/gen_profiles_solution.py

```diff
@@ -587,59 +741,80 @@ def stripNonmatchingAPIs(tree, apiName, actuallyDelete = True):
         createInfo = *pCreateInfo->pCreateInfo;
         pInstanceCreateInfo = &createInfo;
 
-        const detail::VpProfileDesc* pDesc = nullptr;
+        const detail::VpProfileDesc* profile_desc = nullptr;
         if (pCreateInfo->pProfile != nullptr) {
-            pDesc = detail::vpGetProfileDesc(pCreateInfo->pProfile->profileName);
-            if (pDesc == nullptr) return VK_ERROR_UNKNOWN;
+            // Unknown profile
+            profile_desc = detail::vpGetProfileDesc(pCreateInfo->pProfile->profileName);
+            if (profile_desc == nullptr) return VK_ERROR_UNKNOWN;
+
+            // Multiple variants profile, not supported
+            VkBool32 multiple_variants = VK_FALSE;
+            VkResult result = vpHasMultipleVariantsProfile(pCreateInfo->pProfile, &multiple_variants);
+            if (result != VK_SUCCESS || multiple_variants == VK_TRUE) return VK_ERROR_UNKNOWN;
         }
 
         if (createInfo.pApplicationInfo == nullptr) {
-            appInfo.apiVersion = pDesc->minApiVersion;
+            appInfo.apiVersion = profile_desc->minApiVersion;
             createInfo.pApplicationInfo = &appInfo;
         }
 
-        if (pDesc != nullptr && pDesc->pInstanceExtensions != nullptr) {
-            bool merge = (pCreateInfo->flags & VP_INSTANCE_CREATE_MERGE_EXTENSIONS_BIT) != 0;
-            bool override = (pCreateInfo->flags & VP_INSTANCE_CREATE_OVERRIDE_EXTENSIONS_BIT) != 0;
+        const std::vector<VpProfileProperties>& profiles = detail::GatherProfiles(*pCreateInfo->pProfile);
 
-            if (!merge && !override && pCreateInfo->pCreateInfo->enabledExtensionCount > 0) {
-                // If neither merge nor override is used then the application must not specify its
-                // own extensions
-                return VK_ERROR_UNKNOWN;
-            }
+        for (std::size_t profile_index = 0, profile_count = profiles.size(); profile_index < profile_count; ++profile_index) {
+            const char* profile_name = profiles[profile_index].profileName;
+            const detail::VpProfileDesc* profile_desc = detail::vpGetProfileDesc(profile_name);
 
-            detail::vpGetExtensions(pCreateInfo->pCreateInfo->enabledExtensionCount,
-                                    pCreateInfo->pCreateInfo->ppEnabledExtensionNames,
-                                    pDesc->instanceExtensionCount,
-                                    pDesc->pInstanceExtensions,
-                                    extensions, merge, override);
-            {
-                bool foundPortEnum = false;
-                for (size_t i = 0; i < extensions.size(); ++i) {
-                    if (strcmp(extensions[i], VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME) == 0) {
-                        foundPortEnum = true;
-                        break;
+            for (std::size_t caps_index = 0, caps_count = profiles.size(); caps_index < profile_count; ++caps_index) {
```

Comment:
Not seeing where `caps_count` is used

Reply:
Nice catch, I am even wondering how it would possibly work! Oo